pkg/component/operator/web/v0/crawl_website.go

```diff
@@ -70,45 +76,67 @@ func (e *execution) CrawlWebsite(input *structpb.Struct) (*structpb.Struct, erro
 	var mu sync.Mutex
 	pageLinks := []string{}
 
+	// We will have the component timeout feature in the future.
+	// Before that, we initialize the context here.
+	ctx, cancel := context.WithTimeout(context.Background(), 120*time.Second)
+	defer cancel()
+
 	// On every a element which has href attribute call callback
 	// Wont be called if error occurs
 	c.OnHTML("a[href]", func(e *colly.HTMLElement) {
+		mu.Lock()
```

Comment:
```suggestion
		mu.Lock()
                defer mu.Unlock()
```
The mutex need to be unlocked no matter what. We can use `defer` to guard this. I assume the reason you didn’t use `defer` is that you didn’t want to scope `_ = e.Request.Visit(link)` within the mutex. Perhaps we can rearrange the code to avoid this issue while still utilizing `defer`.